File: android/src/main/java/org/maplibre/reactnative/components/layers/types/fill/MLRNFillLayerManager.java
package org.maplibre.reactnative.components.layers.types.fill;

import com.facebook.react.uimanager.ThemedReactContext;

import org.maplibre.reactnative.components.layers.MLRNVectorLayerManager;

public class MLRNFillLayerManager extends MLRNVectorLayerManager<MLRNFillLayer> {
    public static final String REACT_CLASS = "MLRNFillLayer";

    @Override
    public String getName() {
        return REACT_CLASS;
    }

    @Override
    protected MLRNFillLayer createLayerViewInstance(ThemedReactContext reactContext) {
        return new MLRNFillLayer(reactContext);
    }
}

File: android/src/main/java/org/maplibre/reactnative/components/layers/types/fill/MLRNFillLayerManager.kt
package org.maplibre.reactnative.components.layers.types.fill

import com.facebook.react.uimanager.ThemedReactContext
import org.maplibre.reactnative.components.layers.MLRNVectorLayerManager

class MLRNFillLayerManager : MLRNVectorLayerManager<MLRNFillLayer?>() {
    override fun getName(): String {
        return REACT_CLASS
    }

    override fun createLayerViewInstance(reactContext: ThemedReactContext?): MLRNFillLayer {
        return MLRNFillLayer(reactContext)
    }

    companion object {
        const val REACT_CLASS: String = "MLRNFillLayer"
    }
}
